scapy/contrib/ikev2.py

```diff
@@ -313,6 +492,53 @@ class IKEv2_payload_VendorID(IKEv2_class):
         StrLenField("vendorID","",length_from=lambda x:x.length-4),
         ]
 
+class TrafficSelector(IKEv2_class):
```

Comment:
This is ugly.

Please use a dedicated class per traffic selector, and implement `guess_payload_class()` to do the dispatching.